src/tools/generate_insights.py

```diff
@@ -228,143 +179,158 @@ async def generate_insights(
-    if token_count > 23_000:
-        return Command(
-            update={
-                "raw_data": {},  # reset raw data
-                "messages": [
-                    ToolMessage(
-                        content="I've reached my processing limit - you may have requested a large set of areas or too many data points. I'm clearing the current dataset to prevent errors. To continue your analysis, please start a new chat conversation and re-select your areas and datasets.",
-                        tool_call_id=tool_call_id,
-                        status="success",
-                        response_metadata={"msg_type": "human_feedback"},
-                    )
-                ],
-            }
+        dataset_cautions = state.get("dataset").get(
+            "cautions", "No specific dataset cautions provided."
         )
 
-    dataset_guidelines = state.get("dataset").get(
-        "prompt_instructions", "No specific dataset guidelines provided."
-    )
-    dataset_cautions = state.get("dataset").get(
-        "cautions", "No specific dataset cautions provided."
-    )
+        chart_insight_prompt = f"""Based on analysis done by an expert & data saved for visualization, generate structured response.
 
-    try:
-        prompt = _create_insight_generation_prompt()
-        chain = prompt | SONNET.with_structured_output(InsightResponse)
-        response = await chain.ainvoke(
-            {
-                "user_query": query,
-                "raw_data_prompt": raw_data_prompt,
-                "dataset_guidelines": dataset_guidelines,
-                "dataset_cautions": dataset_cautions,
-            }
-        )
+### Analysis Output
+{result.text_output}
 
-        insight = response.insight
-        follow_ups = response.follow_up_suggestions
-        logger.debug(
-            f"Generated insight: {insight.title} ({insight.chart_type})"
-        )
-        logger.debug(f"Generated {len(follow_ups)} follow-up suggestions")
+### Saved chart data - head 5 rows
+{chart_data_df.head().to_csv(index=False)}
 
-        # Format the response message
-        message_parts = []
+Total rows: {len(chart_data_df)}
 
-        message_parts.append(f"Title: {insight.title}")
-        message_parts.append(f"Key Finding: {insight.insight}")
+### Language Context
+Generate ALL content (insights, titles, follow-ups) in the SAME LANGUAGE as the user query.
+Dataset specific guidelines: {dataset_guidelines}
+Dataset specific cautions: {dataset_cautions}
 
-        # Add follow-up suggestions
-        message_parts.append("Follow-up suggestions:")
-        for i, suggestion in enumerate(follow_ups, 1):
-            message_parts.append(f"{i}. {suggestion}")
+#### Capability Context
+You can analyze data for any area of interest, pull data from datasets like {available_datasets} for different time periods, and create various charts/insights. Base follow-ups on what's actually possible with available data and tools.
```

Comment:
this does not make sense to me at this point. We are not pulling data here at all.